manager/container.go

```diff
@@ -331,20 +335,28 @@ func (cd *containerData) GetProcessList(cadvisorContainer string, inHostNamespac
 		if err != nil {
 			return nil, fmt.Errorf("invalid virtual size %q: %v", fields[7], err)
 		}
+		cmd := fields[10]
 		// convert to bytes
 		vs *= 1024
-		psr, err := strconv.Atoi(fields[11])
+		fieldNo := 11
+		var psr int
+		psr, err = strconv.Atoi(fields[fieldNo])
 		if err != nil {
-			return nil, fmt.Errorf("invalid pid %q: %v", fields[1], err)
+			fieldNo++
+			psr, err = strconv.Atoi(fields[fieldNo])
```

Comment:
If I'm understanding the logic here, the idea is that the cmd should be at index 10 and the psr should be at index 11 ASSUMING there is no spaces in the cmd name (this was old logic).

The fix here is basically assume that cmd==index10 and psr==index11, but if you parse as index11 and get error from `strconv.Atoi` it means you have a space in the cmd name. In that case, it means psr is at some greater index than 11 and keep advancing `fieldNo` until you don't get an error from `strconv.Atoi`.

Am I understanding the workaround here correctly? If so, I feel like this a bit fragile... what if the command name has a space and the string after the space is a number, e.g. "mycmd 1". In that case, wouldn't you assume that "1" actually represents psr value, but actually it's just part of the command name and thus parse the output incorrectly?

Reply:
I think that parsing logic has been significantly improved and simplified now :)